src/reset.rs

```diff
@@ -27,6 +27,51 @@ pub enum ResetReason {
     TaskWatchdog,
     /// Reset after exiting deep sleep mode
     DeepSleep,
+    /// Reset by USB peripheral (introduced in IDF v5.1.4)
+    #[cfg(not(any(
+        esp_idf_version_major = "4",
+        all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
+        esp_idf_version_full = "5.1.0",
+        esp_idf_version_full = "5.1.1",
+        esp_idf_version_full = "5.1.2",
```

Comment:
missing 5.1.3 case. Not nice that they made that change on a patch release and not a minor release ;D

Reply:
:facepalm: